airbyte-integrations/connectors/source-chargebee/source_chargebee/streams.py

```diff
@@ -141,54 +197,94 @@ def get_updated_state(self, current_stream_state: MutableMapping[str, Any], late
         latest_cursor_value = latest_record.get(self.cursor_field)
 
         if latest_cursor_value:
-            state_value = {"updated_at": latest_cursor_value}
+            state_value = {self.cursor_field: latest_cursor_value}
 
         return state_value
 
 
-class Subscription(ChargebeeStream):
+class Subscription(IncrementalChargebeeStream):
     """
     API docs: https://apidocs.chargebee.com/docs/api/subscriptions?prod_cat_ver=2#list_subscriptions
     """
 
     api = SubscriptionModel
 
 
-class Customer(ChargebeeStream):
+class Customer(IncrementalChargebeeStream):
     """
     API docs: https://apidocs.chargebee.com/docs/api/customers?prod_cat_ver=2#list_customers
     """
 
     api = CustomerModel
 
 
-class Invoice(ChargebeeStream):
+class Invoice(IncrementalChargebeeStream):
     """
     API docs: https://apidocs.chargebee.com/docs/api/invoices?prod_cat_ver=2#list_invoices
     """
 
     api = InvoiceModel
 
 
-class Order(ChargebeeStream):
+class Order(IncrementalChargebeeStream):
     """
     API docs: https://apidocs.chargebee.com/docs/api/orders?prod_cat_ver=2#list_orders
     """
 
     api = OrderModel
 
 
-class Plan(ChargebeeStream):
+class Plan(IncrementalChargebeeStream):
     """
     API docs: https://apidocs.chargebee.com/docs/api/plans?prod_cat_ver=1&lang=curl#list_plans
     """
 
     api = PlanModel
 
 
-class Addon(ChargebeeStream):
+class Addon(IncrementalChargebeeStream):
     """
     API docs: https://apidocs.chargebee.com/docs/api/addons?prod_cat_ver=1&lang=curl#list_addons
     """
 
     api = AddonModel
+
+
+class Item(IncrementalChargebeeStream):
+    """
+    API docs: https://apidocs.chargebee.com/docs/api/items?prod_cat_ver=2#list_items
+    """
+
+    api = ItemModel
+
+
+class ItemPrice(IncrementalChargebeeStream):
+    """
+    API docs: https://apidocs.chargebee.com/docs/api/item_prices?prod_cat_ver=2#list_item_prices
+    """
+
+    api = ItemPriceModel
+
+
+class AttachedItem(SemiIncrementalChargebeeStream):
+    """
+    API docs: https://apidocs.chargebee.com/docs/api/attached_items?prod_cat_ver=2#list_attached_items
+    """
+
+    api = AttachedItemModel
+
+    def stream_slices(self, **kwargs) -> Iterable[Optional[Mapping[str, Any]]]:
+        items_stream = Item(start_date="")
+        for item in items_stream.read_records(sync_mode=SyncMode.full_refresh):
+            yield {"item_id": item["id"]}
+
+    @default_backoff_handler(max_tries=MAX_TRIES, factor=MAX_TIME)
```

Comment:
backoff should be defined at the top level ideally (it's an API-level concern, not a specific stream's concern), can we adjust this structure so we call super() with kwargs and let super handle the rate limiting?